payroll/models/hr_payslip.py

```diff
@@ -211,8 +214,23 @@ def action_payslip_done(self):
         return self.write({"state": "done"})
 
     def action_payslip_cancel(self):
-        if self.filtered(lambda slip: slip.state == "done"):
-            raise UserError(_("Cannot cancel a payslip that is done."))
+        allow_cancel_payslips = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("payroll.allow_cancel_payslips")
+        )
+        for payslip in self:
+            if allow_cancel_payslips:
+                if payslip.refunded_id and payslip.refunded_id.state != "cancel":
+                    raise ValidationError(
+                        _(
+                            """To cancel the Original Payslip the
+                        Refunded Payslip needs to be canceled first!"""
+                        )
+                    )
+            else:
+                if self.filtered(lambda slip: slip.state == "done"):
+                    raise UserError(_("Cannot cancel a payslip that is done."))
```

Comment:
Does the current behaviour allow you to cancel a payslip that is not in 'Done' state? If so, then I would suggest checking 'allow_cancel_payslips' only if the state is in done.

Reply:
Yes the payslip can be canceled in any state. When the setting is False, payslip in done state can not be canceled. When it's true, they can be canceled.